Build failure: asleap

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
nix-shell -p asleap
```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
this derivation will be built:
  /nix/store/i3nmij62jk3ix0m83g4bkmc5mz7grkqr-asleap-0-unstable-2020-11-28.drv
these 4 paths will be fetched (0.70 MiB download, 3.19 MiB unpacked):
  /nix/store/dpb7blsmabn5sx142nnw2yl1wwwmncjj-libnl-3.11.0
  /nix/store/6hw4aym42f8xfnr6qgn5yss2l15rdn52-libpcap-1.10.5
  /nix/store/5rs376cqh36kx91mafhcx80jw4xfxdf3-libpcap-1.10.5-lib
  /nix/store/3x7jj8lj81zicwsxsqh2jsi3gb7v2wn3-source
copying path '/nix/store/3x7jj8lj81zicwsxsqh2jsi3gb7v2wn3-source' from 'https://cache.nixos.org'...
copying path '/nix/store/dpb7blsmabn5sx142nnw2yl1wwwmncjj-libnl-3.11.0' from 'https://cache.nixos.org'...
copying path '/nix/store/5rs376cqh36kx91mafhcx80jw4xfxdf3-libpcap-1.10.5-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/6hw4aym42f8xfnr6qgn5yss2l15rdn52-libpcap-1.10.5' from 'https://cache.nixos.org'...
building '/nix/store/i3nmij62jk3ix0m83g4bkmc5mz7grkqr-asleap-0-unstable-2020-11-28.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/3x7jj8lj81zicwsxsqh2jsi3gb7v2wn3-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
gcc -pipe -Wall -D_LINUX -D_OPENSSL_MD4 -g3    -c -o sha1.o sha1.c
sha1.c: In function 'SHA1Transform':
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:95:5: note: in expansion of macro 'R0'
   95 |     R0(a, b, c, d, e, 0);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:96:5: note: in expansion of macro 'R0'
   96 |     R0(e, a, b, c, d, 1);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:97:5: note: in expansion of macro 'R0'
   97 |     R0(d, e, a, b, c, 2);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:98:5: note: in expansion of macro 'R0'
   98 |     R0(c, d, e, a, b, 3);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:99:5: note: in expansion of macro 'R0'
   99 |     R0(b, c, d, e, a, 4);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:100:5: note: in expansion of macro 'R0'
  100 |     R0(a, b, c, d, e, 5);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:101:5: note: in expansion of macro 'R0'
  101 |     R0(e, a, b, c, d, 6);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:102:5: note: in expansion of macro 'R0'
  102 |     R0(d, e, a, b, c, 7);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:103:5: note: in expansion of macro 'R0'
  103 |     R0(c, d, e, a, b, 8);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:104:5: note: in expansion of macro 'R0'
  104 |     R0(b, c, d, e, a, 9);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:105:5: note: in expansion of macro 'R0'
  105 |     R0(a, b, c, d, e, 10);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:106:5: note: in expansion of macro 'R0'
  106 |     R0(e, a, b, c, d, 11);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:107:5: note: in expansion of macro 'R0'
  107 |     R0(d, e, a, b, c, 12);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:108:5: note: in expansion of macro 'R0'
  108 |     R0(c, d, e, a, b, 13);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:109:5: note: in expansion of macro 'R0'
  109 |     R0(b, c, d, e, a, 14);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:36:42: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   36 | #define blk0(i) (block->l[i] = (rol(block->l[i], 24) & 0xFF00FF00) | \
      |                                          ^~
sha1.c:31:29: note: in definition of macro 'rol'
   31 | #define rol(value, bits) (((value) << (bits)) | ((value) >> (32 - (bits))))
      |                             ^~~~~
sha1.c:46:32: note: in expansion of macro 'blk0'
   46 |     z += ((w & (x ^ y)) ^ y) + blk0(i) + 0x5A827999 + rol(v, 5); \
      |                                ^~~~
sha1.c:110:5: note: in expansion of macro 'R0'
  110 |     R0(a, b, c, d, e, 15);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:159:5: note: in expansion of macro 'R4'
  159 |     R4(b, c, d, e, a, 64);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:160:5: note: in expansion of macro 'R4'
  160 |     R4(a, b, c, d, e, 65);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:161:5: note: in expansion of macro 'R4'
  161 |     R4(e, a, b, c, d, 66);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:162:5: note: in expansion of macro 'R4'
  162 |     R4(d, e, a, b, c, 67);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:163:5: note: in expansion of macro 'R4'
  163 |     R4(c, d, e, a, b, 68);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:164:5: note: in expansion of macro 'R4'
  164 |     R4(b, c, d, e, a, 69);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:165:5: note: in expansion of macro 'R4'
  165 |     R4(a, b, c, d, e, 70);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:166:5: note: in expansion of macro 'R4'
  166 |     R4(e, a, b, c, d, 71);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:167:5: note: in expansion of macro 'R4'
  167 |     R4(d, e, a, b, c, 72);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:168:5: note: in expansion of macro 'R4'
  168 |     R4(c, d, e, a, b, 73);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:169:5: note: in expansion of macro 'R4'
  169 |     R4(b, c, d, e, a, 74);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:170:5: note: in expansion of macro 'R4'
  170 |     R4(a, b, c, d, e, 75);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:171:5: note: in expansion of macro 'R4'
  171 |     R4(e, a, b, c, d, 76);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:172:5: note: in expansion of macro 'R4'
  172 |     R4(d, e, a, b, c, 77);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:173:5: note: in expansion of macro 'R4'
  173 |     R4(c, d, e, a, b, 78);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
sha1.c:41:22: warning: array subscript 'CHAR64LONG16[0]' is partly outside array bounds of 'unsigned char[64]' [-Warray-bounds=]
   41 | #define blk(i) (block->l[i & 15] = rol(block->l[(i + 13) & 15] ^ \
      |                      ^~
sha1.c:57:24: note: in expansion of macro 'blk'
   57 |     z += (w ^ x ^ y) + blk(i) + 0xCA62C1D6 + rol(v, 5); \
      |                        ^~~
sha1.c:174:5: note: in expansion of macro 'R4'
  174 |     R4(b, c, d, e, a, 79);
      |     ^~
sha1.c:73:48: note: object 'buffer' of size [0, 64]
   73 | void SHA1Transform(u32 state[5], unsigned char buffer[64])
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~
gcc -pipe -Wall -D_LINUX -D_OPENSSL_MD4 -g3    -c -o common.o common.c
common.c: In function 'NtPasswordHash':
common.c:128:5: warning: 'MD4' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  128 |     MD4WRAP(unicodePassword, secret_len * 2, hash);
      |     ^~~~~~~
In file included from common.c:33:
/nix/store/ydrckgnllgg8nmhdwni81h7xhcpnrlhd-openssl-3.6.0-dev/include/openssl/md4.h:53:38: note: declared here
   53 | OSSL_DEPRECATEDIN_3_0 unsigned char *MD4(const unsigned char *d, size_t n,
      |                                      ^~~
gcc -pipe -Wall -D_LINUX -D_OPENSSL_MD4 -g3    -c -o utils.o utils.c
utils.c: In function 'DesEncrypt':
utils.c:179:5: error: implicit declaration of function 'setkey' [-Wimplicit-function-declaration]
  179 |     setkey((char *)crypt_key);
      |     ^~~~~~
utils.c:182:5: error: implicit declaration of function 'encrypt'; did you mean 'crypt'? [-Wimplicit-function-declaration]
  182 |     encrypt((char *)des_input, 0);
      |     ^~~~~~~
      |     crypt
make: *** [<builtin>: utils.o] Error 1
error: Cannot build '/nix/store/i3nmij62jk3ix0m83g4bkmc5mz7grkqr-asleap-0-unstable-2020-11-28.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/fbh3haabd8b98mihws38y4xy32j96py9-asleap-0-unstable-2020-11-28
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.59, NixOS, 26.05 (Yarara), 26.05pre903292.2fad6eac6077`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/nddwf60xb1qav58zjn0jzlxyv94x4bjp-nixos/nixos`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)
@pyrox0 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/as/asleap/package.nix
+++ b/pkgs/by-name/as/asleap/package.nix
@@ -10,13 +10,13 @@
 
 stdenv.mkDerivation {
   pname = "asleap";
-  version = "0-unstable-2020-11-28";
+  version = "0-unstable-2021-06-20";
 
   src = fetchFromGitHub {
-    owner = "joswr1ght";
+    owner = "zackw";
     repo = "asleap";
-    rev = "254acabba34cb44608c9d2dcf7a147553d3d5ba3";
-    hash = "sha256-MQjPup3EX7DCXY/zyroTj/+U2GIq12+VQQJD0gru7C8=";
+    rev = "eb3bd42098cba42b65f499c9d8c73d890861b94f";
+    hash = "sha256-S6jS0cg9tHSfmP6VHyISkXJxczhPx3HDdxT46c+YmE8=";
   };
 
   buildInputs = [
@@ -34,7 +34,9 @@ stdenv.mkDerivation {
     runHook postInstall
   '';
 
-  passthru.updateScript = nix-update-script { extraArgs = [ "--version=branch" ]; };
+  passthru.updateScript = nix-update-script {
+    extraArgs = [ "--version=branch=zack/no-external-crypto" ];
+  };
 
   meta = {
     homepage = "https://github.com/zackw/asleap";

